mava/systems/jax/config.py

```diff
@@ -0,0 +1,139 @@
+        """
+        for name, dataclass in kwargs.items():
+            if is_dataclass(dataclass):
+                if name in list(self._config.keys()):
+                    raise Exception(
+                        "The given component config is already part of the current \
+                        system. Perhaps try updating the component instead using \
+                        .update() in the system builder."
+                    )
+                else:
+                    new_param_names = list(dataclass.__dict__.keys())
+                    if set(self._current_params) & set(new_param_names):
+                        raise Exception(
+                            "Component configs share a common parameter name. \
+                            This is not allowed, please ensure config \
+                            parameter names are unique."
+                        )
+                    else:
+                        self._current_params.extend(new_param_names)
+                    self._config[name] = dataclass
+            else:
+                raise Exception("Component configs must be a dataclass.")
+
+    def update(self, **kwargs: Any) -> None:
+        """Update a component config dataclass.
+
+        Raises:
+            Exception: if a config shares a parameter name with another config
+            Exception: if a config is not already part of the system
+            Exception: if a config is not a dataclass object
+        """
+        for name, dataclass in kwargs.items():
+            if is_dataclass(dataclass):
+                if name in list(self._config.keys()):
+                    new_param_names = list(dataclass.__dict__.keys())
+                    if set(self._current_params) & set(new_param_names):
+                        raise Exception(
+                            "Component configs share a common parameter name. \
+                            This is not allowed, please ensure config \
+                            parameter names are unique."
+                        )
+                    else:
+                        self._current_params.extend(new_param_names)
+                    self._config[name] = dataclass
+                else:
+                    raise Exception(
+                        "The given component config is not part of the current \
+                        system. Perhaps try adding the component using .add() \
+                        in the system builder."
+                    )
+            else:
+                raise Exception("Component configs must be a dataclass.")
+
+    def build(self) -> None:
+        """Build the config file, i.e. unwrap dataclass nested dictionaries"""
+        config_unwrapped: Dict = {}
+        for param in self._config.values():
+            config_unwrapped.update(flatten_dict(param.__dict__))
+        self._built_config = config_unwrapped
+        self._built = True
+
+    def set(self, **kwargs: Any) -> None:
```

Comment:
Can we possibly make this function name a little more descriptive, something like `set_build_config`. This could be confused with the update methods.

Reply:
I will have a bit of a think around updating the names to try to make them more descriptive.

Reply:
I changed the name to `set_parameters`. Let me know if you think this is distinctive enough from `update`.

Reply:
Sounds good :)